### PR TITLE
I Added New Loss Function For Using Object-Detection Task Time.

### PR DESCRIPTION
**Pull request by hisushanta**
This loss function is primarily for working on calculating two bounding how different they are.
Screenshot from 2023-07-01 20-46-52

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @voznesenskym @penguinwu @anijain2305 @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78

### CODE REVIEW

**Comment by hi-sushanta** [comment]
How much time take to review my code?

**Review by mikaylagawarecki — COMMENTED** [review]
Hi @hi-sushanta, could you please file a [feature request](https://github.com/pytorch/pytorch/issues/new?assignees=&labels=&projects=&template=feature-request.yml) detailing the use-cases for this loss function (papers where it is used would be helpful). That would be helpful in driving a discussion of whether this loss function should be added

**Comment by hi-sushanta** [comment]
This is actually a localization loss for using face detection.

**Comment by mikaylagawarecki** [comment]
@hi-sushanta could you please file a [feature request here ](https://github.com/pytorch/pytorch/issues/new?assignees=&labels=&projects=&template=feature-request.yml) with more details (link papers where it is used), thanks!

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -1761,6 +1762,27 @@ def forward(self, log_probs: Tensor, targets: Tensor, input_lengths: Tensor, tar
         return F.ctc_loss(log_probs, targets, input_lengths, target_lengths, self.blank, self.reduction,
                           self.zero_infinity)
 
+
+# This Loss function using for Object Detection Task.
```
Comment (davidradl) [inline torch/nn/modules/loss.py:1766]: the comment might better english as 
This Loss function can be used for Object detection tasks.

Comment (hi-sushanta) [reply torch/nn/modules/loss.py]: I changed my comment. and now look like this : 
Screenshot from 2023-07-07 05-40-33

Comment (hi-sushanta) [reply torch/nn/modules/loss.py]: Ok now add a full description of my custom loss function.

look like this : 
Screenshot from 2023-07-07 09-27-34

If it's fine.

File: torch/nn/modules/loss.py [hunk 2]
```diff
@@ -1761,6 +1762,63 @@ def forward(self, log_probs: Tensor, targets: Tensor, input_lengths: Tensor, tar
...
 
+
+# This Loss function can be used for Object detection tasks.
+class OBJLoss(_Loss):
+    def __init__(self):
+        super(OBJLoss, self).__init__()
+
+    def forward(self, y_true:Tensor, yhat:Tensor):
+        """
+        OBJLoss is a custom loss function for object detection. It is a combination of the coordinate loss and the size loss.
```
[the remark this answers is not in the document]
Comment (hi-sushanta) [reply torch/nn/modules/loss.py]: I rename the OBJLoss to ObjectDetectionCoordinatesAndSizeLoss class.

[the remark this answers is not in the document]
Comment (hi-sushanta) [reply torch/nn/modules/loss.py]: If you see my changes?

**Comment by hi-sushanta** [comment]
I submit CLA file but status don't update.
